homeassistant/components/remote/harmony.py

```diff
@@ -0,0 +1,163 @@
+"""
+Support for Harmony Hub devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/remote.harmony/
+
+"""
+
+import logging
+from homeassistant.const import (
+    CONF_NAME, CONF_HOST, CONF_PORT)
+from homeassistant.components.remote import PLATFORM_SCHEMA
+from homeassistant.util import slugify
+import homeassistant.components.remote as remote
+import homeassistant.helpers.config_validation as cv
+import voluptuous as vol
+
+REQUIREMENTS = ['pyharmony==1.0.12']
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_DEVICE = 'device'
+ATTR_COMMAND = 'command'
+ATTR_ACTIVITY = 'activity'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PORT): cv.string,
+    vol.Required(ATTR_ACTIVITY, default='Not_Set'): cv.string,

```

Comment:
This is a weird default. Why not have it be `None` and at the point where you actually need a not set value replace it ? Otherwise you can no longer differentiate between an omitted value or not .